src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,648 @@
+        r"""
+        Return the fundamental group of the complement of an affine
+        hyperplane arrangement in `\CC^n` whose equations have
+        coefficients in a subfield of `\QQbar`.
+
+        OUTPUT:
+
+        A finitely presented fundamental group`.
+
+        .. NOTE::
+
+           This functionality requires the ``sirocco`` package to be installed.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: A.<x, y> = OrderedHyperplaneArrangements(QQ)
+            sage: L = [y + x, y + x - 1]
+            sage: H = A(L)
+            sage: H.affine_fundamental_group()
+            Finitely presented group < x0, x1 |  >
+            sage: L = [x, y, x + 1, y + 1, x - y]
+            sage: A(L).affine_fundamental_group()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4*x0*x4^-1*x0^-1,
+                                   x0*x2*x3*x2^-1*x0^-1*x3^-1,
+                                   x1*x2*x4*x2^-1*x1^-1*x4^-1,
+                                   x2*x3*x0*x2^-1*x0^-1*x3^-1,
+                                   x2*x4*x1*x2^-1*x1^-1*x4^-1,
+                                   x4*x1*x4^-1*x3^-1*x2^-1*x1^-1*x2*x3 >
+            sage: H = A(x, y, x + y)
+            sage: H.affine_fundamental_group()
+            Finitely presented group
+            < x0, x1, x2 | x0*x1*x2*x1^-1*x0^-1*x2^-1, x1*x2*x0*x1^-1*x0^-1*x2^-1 >
+            sage: H.affine_fundamental_group()  # repeat to use the attribute
+            Finitely presented group
+            < x0, x1, x2 | x0*x1*x2*x1^-1*x0^-1*x2^-1, x1*x2*x0*x1^-1*x0^-1*x2^-1 >
+            sage: T.<t> = QQ[]
+            sage: K.<a> = NumberField(t^3 + t + 1)
+            sage: L.<x, y> = OrderedHyperplaneArrangements(K)
+            sage: H = L(a*x + y -1, x + a*y + 1, x - 1, y - 1)
+            sage: H.affine_fundamental_group()
+            Traceback (most recent call last):
+            ...
+            TypeError: the base field is not in QQbar
+            sage: L.<t> = OrderedHyperplaneArrangements(QQ)
+            sage: L([t - j for j in range(4)]).affine_fundamental_group()
+            Finitely presented group < x0, x1, x2, x3 |  >
+            sage: L.<x, y, z> = OrderedHyperplaneArrangements(QQ)
+            sage: L(L.gens() + (x + y + z + 1,)).affine_fundamental_group().sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3 | x3^-1*x2^-1*x3*x2, x3^-1*x1^-1*x3*x1,
+                               x3^-1*x0^-1*x3*x0, x2^-1*x1^-1*x2*x1,
+                               x2^-1*x0^-1*x2*x0, x1^-1*x0^-1*x1*x0 >
+        """
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        if self._affine_fundamental_group:
+            return self._affine_fundamental_group
+        n = self.dimension()
+        r = len(self)
+        if n == 0:
+            raise TypeError('Only empty arrangements for dimension ', n)
```

Comment:
This seems wrong. Please include a test for it.

Also, error messages should start with a lower case as well follow Python's convention.

Reply:
I've changed it to return the trivial group. It was a (probably unneeded) precaution (next push).